### PR TITLE
Consolidate Signature and Member content generation code

### PR DESCRIPTION
**Pull request by BZngr**
Refactorings that generate code (e.g., `ImplementInterface, EncapsulateField`, and others) have a common interest in creating blocks of code for Properties and Methods based on, or using, existing Member and Variable declarations.  This common interest has resulted in similar (if not identical) code showing up in multiple refactorings and in UI elements as well (e.g., just to display the signature).  So, as I was pondering how to avoid duplicating the code in a fourth location (`MoveMemberRefactoring`), it seemed that it was time to attempt a PR that would consolidate this functionality in a central location.

In its current state, the PR introduces this functionality through `DeclarationExtension` and `ModuleBodyElementDeclarationExtension`  methods in the `Rubberduck.Refactorings.Common` folder.  This approach/location can certainly be changed.  The primary goal was to corral this capability into a single place where it could be easily reused by the refactorings.  And, I figured it would be easier for all to evaluate where this code belongs if there was a working version from which to start.

Notes:
1.  The PR primarily affects the `EncapsulateFieldRefactoring` and `AddInterfaceImplementationsRefactoringAction` refactorings.

2. There is a UI converter class `DeclarationToMemberSignatureConverter` which also duplicates the code to build a full member signature from the `Declaration`.  It seems that the UI project does not have access to the `Rubberduck.Refactorings.Common` namespace(?) - so for now, it is left as-is.
**Update**: I am unable to find any references to `DeclarationToMemberSignatureConverter` in the xaml files (or anywhere else).  I think the file/class can be deleted.

3. The functions in `DeclarationExtensions.cs` for this PR begin with `FieldToPropertyBlock` near the end of the file.  There are other functions starting at the top of the file that were moved there from `EncapulateFieldRefactoring`.  A few are relevant to support this PR.  I've moved the others there in preparation for the `MoveMemberRefactoring`.

### CODE REVIEW

**Comment by BZngr** [comment]
Merging this PR should wait for the same reason(s) as given for #5420

**Comment by retailcoder** [comment]
It's been 14 days, I'm not going to hold up merging this PR - I'll just deal with any conflicts on my side =)

File: Rubberduck.Refactorings/Common/DeclarationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        /// Generates a Property Member code block specified by the letSetGet DeclarationType argument.
+        /// </summary>
+        /// <param name="variable"></param>
+        /// <param name="letSetGetType"></param>
+        /// <param name="propertyIdentifier"></param>
+        /// <param name="accessibility"></param>
+        /// <param name="content"></param>
+        /// <param name="parameterIdentifier"></param>
+        /// <returns></returns>
+        public static string FieldToPropertyBlock(this Declaration variable, DeclarationType letSetGetType, string propertyIdentifier, string accessibility = null, string content = null, string parameterIdentifier = null)
```
Comment (retailcoder) [inline Rubberduck.Refactorings/Common/DeclarationExtensions.cs]: I'm not sure this type of work really belongs here as an extension method to `Declaration`... would the target object not always be a `VariableDeclaration`?

File: Rubberduck.Refactorings/Common/ModuleBodyElementDeclarationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    public static class ModuleBodyElementDeclarationExtensions
+    {
+        /// <summary>
+        /// Returns ModuleBodyElementDeclaration signature with an ImprovedArgument list
+        /// 1. Explicitly declares Property Let\Set value parameter as ByVal
+        /// 2. Ensures UserDefined Type parameters are declared either explicitly or implicitly as ByRef
+        /// </summary>
+        /// <param name="declaration"></param>
+        /// <returns></returns>
+        public static string FullMemberSignature(this ModuleBodyElementDeclaration declaration,
```
Comment (retailcoder) [inline Rubberduck.Refactorings/Common/ModuleBodyElementDeclarationExtensions.cs]: :+1: for not extending `Declaration` (and thus not hiding the coupling) - note, there's something funny with extension methods that extend a specific concrete implementation: normally you'd write extension methods against an interface (and treat all implementations equally).
I guess what I'm trying to say is, there's probably a better design, but I can't think of one at the moment, and this works, so... moving on =)

File: Rubberduck.Refactorings/Common/DeclarationExtensions.cs (hunk 1 above)
Comment (BZngr) [reply Rubberduck.Refactorings/Common/DeclarationExtensions.cs]: Yes - should have been `VariableDeclaration`.  Once the dust settles surrounding `EncapsulateField` - I'll put together a PR for this.

Comment (MDoerner) [reply Rubberduck.Refactorings/Common/DeclarationExtensions.cs]: I think this does not belong into an extension at all. Instead, I thing there should be some `ICodeBulder` to build the standard code for declarations. That would make the standard easily testable and would also allow to specify different versions of the code by injecting another implementation.
The same applies to the extensions for module body elements, even more, actually.

Comment (BZngr) [reply Rubberduck.Refactorings/Common/DeclarationExtensions.cs]: @MDoerner I was certain using an extension was a debatable location for this functionality - and I was hoping for either confirmation or some guidance to move it elsewhere - thanks.  As I understand your comment, the `ICodeBuilder`  instance would be delivered by CW? 

 As per my earlier comment, I was planning to wait for the coming `EncapsulateField` PR before introducing anything further changes.

File: Rubberduck.Resources/RubberduckUI.resx [hunk 3]
```diff
@@ -1779,4 +1779,7 @@ Import aborted.</value>
   <data name="EncapsulateField_WrapFieldsInPrivateType" xml:space="preserve">
     <value>Wrap Fields in Private Type</value>
   </data>
+  <data name="ImplementInterface_TODO" xml:space="preserve">
+    <value>'TODO implement interface member</value>
```
Comment (MDoerner) [inline Rubberduck.Resources/RubberduckUI.resx]: This does not belong here; it is not related to the UI.

**Pull request merged**